packages/vega-parser/src/parsers/guides/axis-title.js

```diff
@@ -33,14 +35,31 @@ export default function(spec, config, userEncode, dataRef) {
     signal: `lerp(range("${spec.scale}"), ${anchorExpr(0, 1, 0.5)})`
   };
 
-  if (horizontal) {
-    update.x = titlePos;
-    enter.angle = {value: 0};
-    enter.baseline = {value: orient === Top ? 'bottom' : 'top'};
+  if (isSignal(orient)) {
+    update.x = xyAxisConditionalEncoding('x', orient.signal, titlePos, null);
+    update.y = xyAxisConditionalEncoding('y', orient.signal, titlePos, null);
+    enter.angle = update.angle = 
+      xyAxisConditionalEncoding('x',
+        orient.signal,
+        zero,
+        { signal: `(${sign.signal}) * 90` }
+      );
+    enter.baseline = update.baseline =
+      xyAxisConditionalEncoding('x', 
+        orient.signal,
+        {signal: `(${orient.signal}) === "${Top}" ? "bottom" : "top"`},
+        { value: 'bottom' }
+      );
   } else {
-    update.y = titlePos;
-    enter.angle = {value: sign * 90};
-    enter.baseline = {value: 'bottom'};
+    if (horizontal) {
+      update.x = titlePos;
+      enter.angle = {value: 0};
```

Comment:
why do we use `zero` up there in the signal and `{value: 0}` here? 

I guess they are the same, but better be consistent?

Reply:
We should use `zero` when possible.

Reply:
fixed